### PR TITLE
slack-bot: add context to job links

### PR DESCRIPTION
**Pull request by stevekuznetsov**
When a user pastes a link to a job or artifact, we often want to see the
full Spyglass link and the ci-operator configuration that gave rise to
the job in the first place. Instead of asking them for this information
we can just calculate it and post a reply.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

### CODE REVIEW

File: pkg/slack/events/joblink/link.go [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+
+// NewJobGetter exposes a simplified interface from the Prow configuration
+func NewJobGetter(config config.Getter) JobGetter {
+	return &getter{
+		config: config,
+	}
+}
+
+type getter struct {
+	config config.Getter
```
Comment (alvaroaleman) [inline pkg/slack/events/joblink/link.go:53]: Nit: You can alias the config.Getter type and define methods on the alias

File: pkg/slack/events/joblink/link.go [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+		Repo:    parts[1],
+		Branch:  rehearse.BranchFromRegexes(brancher.Branches),
+		Variant: rehearse.VariantFromLabels(job.Labels),
+	}
+}
+
+func (g *getter) JobForName(name string) *job {
+	var found job
+
+	for orgRepo, presubmits := range g.config().PresubmitsStatic {
```
Comment (alvaroaleman) [inline pkg/slack/events/joblink/link.go:72]: I don't think the name is enough, I think we have the same name on different branches

File: pkg/slack/events/joblink/link.go [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+	case "prow.ci.openshift.org":
+		switch strings.Split(url.Path, "/")[1] {
+		case "job-history":
+			return infoForJobHistory(url)
+		case "log":
+			return infoForJobLog(url)
+		case "view":
+			return infoForJobView(url)
+		}
+	case "gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com":
```
Comment (alvaroaleman) [inline pkg/slack/events/joblink/link.go:319]: can we add this url to the api package so we don't break the tool when we give it a nicer one?

File: pkg/slack/events/joblink/link.go [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+				} `json:"elements"`
+			} `json:"elements"`
+		} `json:"blocks"`
+	}
+	var data blocks
+	fmt.Println(string(*raw.InnerEvent))
+	if err := json.Unmarshal(*raw.InnerEvent, &data); err != nil {
+		return nil, errors.New("could not get blocks from event data")
+	}
+	fmt.Println(data)
```
Comment (bbguimaraes) [inline pkg/slack/events/joblink/link.go:289]: Debug statements.

File: pkg/slack/events/joblink/link.go (hunk 2 above)
Comment (stevekuznetsov) [reply pkg/slack/events/joblink/link.go:72]: Not for generated jobs. I'm pretty sure name is unique in our config.

File: pkg/slack/events/joblink/link.go (hunk 1 above)
Comment (stevekuznetsov) [reply pkg/slack/events/joblink/link.go:53]: :exploding_head:

Comment (stevekuznetsov) [reply pkg/slack/events/joblink/link.go:53]: `Cannot define methods on non-local type config.Getter`

**Comment by stevekuznetsov** [comment]
@alvaroaleman @bbguimaraes updated

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**